File: src/main/java/club/xiaojiawei/enums/CardTypeEnum.java
package club.xiaojiawei.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

/**
 * @author 肖嘉威
 * @date 2022/11/27 19:17
 */
@Getter
@ToString
@AllArgsConstructor
public enum CardTypeEnum {
    MINION("随从"),
    LOCATION("地标"),
    SPELL("法术"),
    HERO("英雄"),
    WEAPON("武器"),
    HERO_POWER("技能"),
    ENCHANTMENT("效果"),
    GAME("游戏"),
    PLAYER("玩家"),
    INVALID("无效"),
    UNKNOWN("未知"),
    ;
    private final String comment;

}

File: Hearthstone-Script-Card-SDK/src/main/kotlin/club/xiaojiawei/enums/CardTypeEnum.kt
package club.xiaojiawei.enums

/**
 * @author 肖嘉威
 * @date 2022/11/27 19:17
 */
enum class CardTypeEnum {
    /**
     * 随从
     */
    MINION,

    /**
     * 地标
     */
    LOCATION,

    /**
     * 法术
     */
    SPELL,

    /**
     * 英雄
     */
    HERO,

    /**
     * 武器
     */
    WEAPON,

    /**
     * 技能
     */
    HERO_POWER,

    /**
     * 效果
     */
    ENCHANTMENT,

    /**
     * 游戏
     */
    GAME,

    /**
     * 玩家
     */
    PLAYER,

    /**
     * 无效
     */
    INVALID,

    /**
     * 未知
     */
    UNKNOWN,
    ;

    companion object {
        fun fromString(str: String?): CardTypeEnum {
            if (str == null || str.isBlank()) return UNKNOWN
            return try {
                valueOf(str.uppercase())
            }catch (_:Exception){
                UNKNOWN
            }
        }
    }
}
